### PR TITLE
Convert checkbox, radio and selectboxes to NL Design System components

### LINKED ISSUE #465: Refactor radio and checkbox components
Currently the CSS imports for these Utrecht NL DS components are missing from our bundle too.

The markup and own CSS is currently not suitable for the Utrecht theme/design tokens. This should just be re-structured to fit the proper component structure.

NOTE: upstream discovered a line-wrapping issue for long label texts, so we should probably hold off on this until that's fixed.

NOTE: check the alignment, our flexbox container is centering everything

### LINKED ISSUE #468: Convert selectboxes to use NL DS component structure
Probably similar to #465, but the selectboxes component does not apply proper markup or styles to use the utrecht checkbox component.

### LINKED ISSUE #476: Refactor pure react checkbox components to NL DS
Leftovers from #475

- [x] Refactor `PrivacyCheckbox`
- [x] Remove `_choices.scss` styles related to checkbox/radio

### PR DESCRIPTION
**Pull request by sergei-maertens**
Closes #465
Closes #468
Closes #476

1. Adding React RadioField broke the Form.io radio component
2. Fixing the Form.io radio component broke the Form.io selectboxes component
3. Fixing the the selectboxes component broke the checkbox

There are some left-over React checkboxes in other components that are hopefully not affected by this (Chromatic will tell...). Once those are converted, we can remove a bunch of CSS again.

### CODE REVIEW

**Comment by sergei-maertens** [comment]
There's a known issue with block-elements (`p` from WYSIWYG) in the checkbox label + the required asterisk, but we'll see if we can tackle that as part of open-formulieren/open-forms#3203

File: src/components/forms/RadioField/RadioField.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import {HelpText, ValidationErrors, Wrapper} from 'components/forms';
+import {LabelContent} from 'components/forms/Label';
+
+import './RadioField.scss';
+
+/**
+ * A radio field with a set of options.
+ *
+ * @reference https://nl-design-system.github.io/utrecht/storybook-react/?path=/docs/react-component-form-field-radio-group--docs
+ */
```
Comment (CharString) [inline src/components/forms/RadioField/RadioField.js:22]: What makes the Utrecht radio group not reusable? I'm thinking of an adaptor component that handles  our Formik state and the prop shape differences between the API of RadioField and the Utrecht Component?

File: src/formio/templates/field.ejs [hunk 2]
```diff
@@ -1,19 +1,55 @@
...
-{% } %}
+      <legend class="utrecht-form-fieldset__legend utrecht-form-fieldset__legend--html-legend utrecht-form-field__label">
+        <span class="utrecht-form-label utrecht-form-label--openforms-required">
+          {{ ctx.t(ctx.component.label, { _userInput: true }) }}
+          {% if (!ctx.component.validate.required && !ctx.requiredFieldsWithAsterisk) { %}
+            {{ ctx.t('(optional)') }}
+          {% } %}
+          {% if (ctx.component.tooltip) { %}
+            <i ref="tooltip" class="{{ctx.iconClass('question-sign')}}"></i>
+          {% } %}
```
Comment (CharString) [inline src/formio/templates/field.ejs:17]:
```suggestion
          {% if (ctx.component.tooltip) { %}
            <i ref="tooltip" class="{{ctx.iconClass('question-sign')}} data-tooltip="{{ ctx.tooltip }}"></i>
          {% } %}
```

File: src/formio/templates/field.ejs [hunk 3]
```diff
@@ -1,19 +1,55 @@
...
 
-{% if (!ctx.label.hidden && ctx.label.labelPosition === 'bottom') { %}
-  {{ ctx.labelMarkup }}
-{% } %}
+      {% if (ctx.component.description) { %}
+        <div
+          class="utrecht-form-field-description utrecht-form-field-description--openforms-helptext"
+          id="{{ctx.id}}-description"
+        >
+          {{ctx.t(ctx.component.description)}}
```
Comment (CharString) [inline src/formio/templates/field.ejs:28]: `ctx.component.description` is already translated server side to handle form variable interpolation. This probably ends up being a no-op. But there's a minute chance double translations may happen. Like the translation chain: streepje => bar => café.

File: src/formio/templates/field.ejs [hunk 4]
```diff
@@ -1,19 +1,55 @@
...
+  {{ctx.element}}
+
+  {% if (!ctx.label.hidden && ctx.label.labelPosition === 'bottom') { %}
+    {{ ctx.labelMarkup }}
+  {% } %}
 
-{% if (ctx.component.description) { %}
-  <div class="utrecht-form-field-description utrecht-form-field-description--openforms-helptext">{{ctx.t(ctx.component.description)}}</div>
+  {% if (ctx.component.description) { %}
+    <div class="utrecht-form-field-description utrecht-form-field-description--openforms-helptext">{{ctx.t(ctx.component.description)}}</div>
```
Comment (CharString) [inline src/formio/templates/field.ejs:52]: Same here

File: src/components/forms/Label.js [hunk 5]
```diff
@@ -6,14 +6,15 @@ import {FormattedMessage} from 'react-intl';
...
-export const LabelContent = ({id, disabled = false, isRequired = false, children}) => {
+export const LabelContent = ({id, disabled = false, isRequired = false, type, children}) => {
   const {requiredFieldsWithAsterisk} = useContext(ConfigContext);
   return (
     <FormLabel
       htmlFor={id}
       disabled={disabled}
       className={classNames({
         'utrecht-form-label--openforms-required': isRequired,
+        [`utrecht-form-label--${type}`]: type,
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/components/forms/Label.js:17]: and it's only Monday!

File: src/components/forms/RadioField/RadioField.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  label: PropTypes.string,
+  isRequired: PropTypes.bool,
+  description: PropTypes.node,
+  id: PropTypes.string,
+  disabled: PropTypes.bool,
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.string.isRequired,
+      label: PropTypes.string.isRequired,
+    })
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/components/forms/RadioField/RadioField.js:92]: but it shouldn't be from an accessibility perspective - you should use proper, explicit labels for options

File: src/components/forms/RadioField/RadioField.js (hunk 1 above)
Comment (sergei-maertens) [reply src/components/forms/RadioField/RadioField.js:22]: I'm not sure what you mean, this is following the example from the utrecht radio group :thinking:

File: src/formio/templates/field.ejs (hunk 3 above)
Comment (sergei-maertens) [reply src/formio/templates/field.ejs:28]: good catch, I just copy pasted it around but let's patch this out.

File: src/components/forms/RadioField/RadioField.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/components/forms/RadioField/RadioField.js:92]: checked - formio itself does not support those variants, it must be something we do in the backend

File: src/components/forms/RadioField/RadioField.js (hunk 1 above)
Comment (sergei-maertens) [reply src/components/forms/RadioField/RadioField.js:22]: clarified - there's no magic component that I missed and Chris caught

**Pull request merged**